packages/analytics-js/src/components/configManager/ConfigManager.ts

```diff
@@ -1,21 +1,148 @@
+/* eslint-disable class-methods-use-this */
+import { IHttpClient } from '@rudderstack/analytics-js/services/HttpClient/types';
+import { IErrorHandler } from '@rudderstack/analytics-js/services/ErrorHandler/types';
+import { ILogger } from '@rudderstack/analytics-js/services/Logger/types';
+import { defaultLogger } from '@rudderstack/analytics-js/services/Logger';
+import { defaultErrorHandler } from '@rudderstack/analytics-js/services/ErrorHandler';
+import { defaultHttpClient } from '@rudderstack/analytics-js/services/HttpClient';
+import { batch } from '@preact/signals-core';
+import { validateLoadArgs } from '@rudderstack/analytics-js/components/configManager/util/validate';
 import { state } from '@rudderstack/analytics-js/state';
+import { Destination } from '@rudderstack/analytics-js/state/types';
+import { resolveDataPlaneUrl } from './util/dataPlaneResolver';
+import { getIntegrationsCDNPath } from './util/cdnPaths';
+import { getSDKUrlInfo } from './util/commonUtil';
+import { IConfigManager, SourceConfigResponse } from './types';
+import { filterEnabledDestination } from './util/filterDestinations';
 
-class ConfigManager {
+class ConfigManager implements IConfigManager {
+  httpClient: IHttpClient;
+  errorHandler?: IErrorHandler;
+  logger?: ILogger;
+  hasErrorHandler = false;
+  hasLogger = false;
+
+  constructor(httpClient: IHttpClient, errorHandler?: IErrorHandler, logger?: ILogger) {
+    this.errorHandler = errorHandler;
+    this.logger = logger;
+    this.httpClient = httpClient;
+    this.hasErrorHandler = Boolean(this.errorHandler);
+    this.hasLogger = Boolean(this.logger);
+  }
+
+  /**
+   * A function to validate, construct and store loadOption, lifecycle, source and destination
+   * config related information in global state
+   */
   init() {
-    // TODO: get writekey, dataplaneUrl and loadOptions from state to use
+    validateLoadArgs(state.lifecycle.writeKey.value, state.lifecycle.dataPlaneUrl.value);
+    // determine the sourceConfig url
+    const intgCdnUrl = getIntegrationsCDNPath(
+      'process.package_version',
+      state.loadOptions.value.lockIntegrationsVersion as boolean,
```

Comment:
This typecasting shouldn't happen here. They have to be pre-validated.

Reply:
It's better to only accept Boolean values for the input. Hence, we can validate them accordingly.

Reply:
done